src/main/java/utils/ConfigReader.java

```diff
@@ -1,19 +1,55 @@
 package utils;
 
 import java.io.FileInputStream;
-import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.util.Properties;
 
 public class ConfigReader {
-    private Properties properties;
 
-    public ConfigReader(String configFileName) throws IOException {
-        properties = new Properties();
-        properties.load(new FileInputStream("src/main/resources/config/" + configFileName));
+    private Properties commandNameToClass = new Properties();
+    private Properties appConfig = new Properties();
+    private Properties arangoConfig = new Properties();
+
+    private volatile static ConfigReader instance;
+
+    public static boolean isTesting;
+
+    private ConfigReader() throws IOException {
+        populateWithConfig("commands.config", commandNameToClass);
+        populateWithConfig(isTesting ? "arango.test.config" : "arango.config", arangoConfig);
+        populateWithConfig("app.config", appConfig);
+    }
+
+    private static void populateWithConfig(String configFileName, Properties properties) throws IOException {
+        String configFolder = "src/main/resources/config/";
+        FileInputStream inputStream = new FileInputStream(configFolder + configFileName);
+        properties.load(inputStream);
+        inputStream.close();
+    }
+
+    public static ConfigReader getInstance() throws IOException {
+        if(instance == null) {
+            synchronized (ConfigReader.class) {
+                if(instance == null)
+                    instance = new ConfigReader();
+            }
+        }
+
+        return instance;
+    }
+
+    public Class getCommandClass(String commandName) throws ClassNotFoundException {
+        String commandsPackageName = appConfig.getProperty("package.commands");
+        String commandClass = commandsPackageName + '.' + commandNameToClass.get(commandName);
+        return Class.forName(commandClass);
+    }
+
+    public Class getNoSqlHandler() throws ClassNotFoundException {
```

Comment:
I think we should not restrict a command to a database type. What if an administrator want to migrate from no sql to a relational one or a graph one. I have an idea we can add to configuration file of commands the number of handlers needed by a command then the handlers name. For example: 

`notifier.getNotification.command=GetNotificationCommand
  notifier.getNotification.handlers.num=2             
  notifier.getNotification.dbHandler=ArangoHandler
  notifier.getNotification.dbHandler=MySqlHandler`

Do you think this is a correct approach?